src/applications/differential/query/DifferentialRevisionRequiredReviewerActionResultBucket.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+final class DifferentialRevisionRequiredReviewerActionResultBucket
+  extends DifferentialRevisionResultBucket {
+
+  const BUCKETKEY = 'review-action';
+
+  const KEY_REJECTED = 'rejected-review';
+  const KEY_BLOCKING = 'blocking-review';
+  const KEY_SHOULDREVIEW = 'should-review';
+
+  private $objects;
+
+  public function getResultBucketName() {
+    return pht('Bucket by Reviewer Required Action');
+  }
+
+  protected function buildResultGroups(
+    PhabricatorSavedQuery $query,
+    array $objects) {
+
+    $this->objects = $objects;
+
+    $phids = $query->getEvaluatedParameter('reviewerPHIDs');
```

Comment:
Call it `$reviewerPhids`?

Also, how does this work if there are multiple reviewers? Does it make sense at all?

Can we get the current user ID if we don't have a reviewers parameter?

Reply:
Im following the same code conventions that other phabricator query and result buckets have so reluctant to change this away from $phids. Yes the the current viewer ID is got by $this->getViewer()->getPHID()